apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/DmlTests.cs

```diff
@@ -535,127 +542,192 @@ public void PartitionedUpdate_Medium()
             }
         }
 
-        [Fact(Skip = "This test is skipped until DML Returning changes are done.")]
-        public async Task DMLReturn_ExecuteReader_Read()
+        // With DML's return clause ExecuteReaderAsync should return a reader containing affected rows data.
+        [Theory, CombinatorialData]
+        [Trait(Constants.SupportedOnEmulator, Constants.No)]
+        public async Task DMLReturn_ExecuteReader_Read(TransactionType transactionType)
         {
             string key = _fixture.CreateTestRows();
-            var actualValues = new List<int>();
-
             using var connection = _fixture.GetConnection();
             await connection.OpenAsync();
-            RetryHelpers.ExecuteWithRetry(async () =>
+            string dml = $"UPDATE {_fixture.TableName} SET Value = OriginalValue + 1 WHERE UpdateMe AND Key=@key Then Return Value";
+            var command = connection.CreateDmlCommand(dml);
+            command.Parameters.Add("key", SpannerDbType.String, key);
+
+            List<int> actualReturnedValues = await Execute(DmlAsyncWork, connection, command, transactionType);
+
+            // Assert that the DML Return command has returned expected values only.
+            var expectedReturnedValues = new List<int> { 2, 5 };
+            Assert.Equal(expectedReturnedValues, actualReturnedValues);
+
+            // Assert that the DML Return command has actually updated the table.
+            AssertActualAndUpdatedDbValues(key);
+
+            async Task<List<int>> DmlAsyncWork(SpannerTransaction retryableTransaction = null)
```

Comment:
This might or might not be a retriable transaction.
```suggestion
            async Task<List<int>> DmlAsyncWork(SpannerTransaction transaction = null)
```